portfolio/src/main/java/com/google/sps/servlets/SalaryDataServlet.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns major-to-salary data as a JSON object, e.g. {"Computer Science": 52000, "Computer Engineering": 55000}] */
+@WebServlet("/salary-data")
+public class SalaryDataServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final LinkedHashMap<String, Integer> majorSalaryData = new LinkedHashMap<>();
```

Comment:
I assume you're using LinkedHashMap over a regular HashMap here to preserve the original ordering?

If so, the JSON spec notes that object keys are unordered, and this is also true of the default Map implementations in many languages. So this is a bit fragile -- it's possible it'll work just fine, but that would only be a side-effect of implementation details (i.e. that Gson preserves iteration order when writing Maps to JSON objects, that the client-side Javascript preserves order when deserializing JSON objects, etc.).

If order is a strict requirement, I'd recommend either using a List<MajorSalaryData> that gets serialized to a JSON array of JSON objects. And if it's not, I'd switch to a regular `HashMap`. You could also just add sorting on the client side so that the client assumes less about the protocol.